### DISCUSSION TITLE
Coalesce macro

### DISCUSSION THREAD

**Discussion by SleeplessOne1917** [diesel-rs/diesel#3884] [Ideas]
Currently, the most straightforward way to use the coalesce function is to create an ad hoc function with the arguments you need using the `sql_function` macro. This can be cumbersome if you need to coalesce expressions with varying numbers or parameters.

Coalesce could likely be variadic if made as a macro. I will attempt to implement this and share my results.

**Comment by weiznich** [comment]
You already can support different kinds of parameters by using generics in combination with `sql_function!`. 

I don't think that macros will solve the variadic case as you somehow need to produce a specific struct in the end. The usual rust "solution" for this kind of problem is to accept a tuple as argument. I think I would be open to extend the `sql_function!` macro to support something like that if someone is willing to spend the time for designing and implementing (including tests + documentation) for such a feature.
